malloc/malloc.c

```diff
@@ -61,29 +61,25 @@ void mem_copy(void *dest, const void *src, int n)
 */
 void *resize_memory(void *ptr, int old_size, int new_size)
 {
-    printf("Resize ");
+
     char *cast_ptr = (char *)ptr;
-    char *new_mem = malloc( new_size * sizeof(char));   
+    
     int size = new_size;
 
     if ( new_size > old_size) {
-
      size = old_size;
-
     }
- printf("Before For ");
+    char *new_mem = malloc( size * sizeof(char));   
+
  
-    for (int i; i < size; i++){
+    for (int i = 0; i < size; i++){
         printf("In For %c\n", *(cast_ptr + i));
        *(new_mem + i) = *(cast_ptr + i);
         }
     
```

Comment:
once you created the new_mem, ptr becomes obsolete, so you should free up the memory space for ptr